src/mappings/mappingHandlers.ts

```diff
@@ -208,6 +208,72 @@ export async function handleCW20Transfer(event: CosmosEvent): Promise<void> {
   await cw20transfer.save();
 }
 
+export async function handleCW20BalanceBurn(event: CosmosEvent): Promise<void> {
+  const id = messageId(event.msg);
+  logger.info(`[handleCW20BalanceBurn] (tx ${event.tx.hash}): indexing CW20BalanceBurn ${id}`);
+
+  const msg = event.msg.msg.decodedMsg;
+  const fromAddress = msg.sender, contract = msg.contract;
+  const amount = msg.msg?.burn?.amount;
+
+  if (typeof(amount)==="undefined" || typeof(fromAddress)==="undefined" || typeof(contract)==="undefined") {
```

Comment:
I don't think we want the strictness of the `typeof(object) === "undefined` here, as opposed to just `!object`. The difference between the two is that while the former is only true if `object` was `undefined`, the latter is true not only if `object` was `undefined` but also if it was `false`, `null`, `0`, or `""` (all falsy in JS). Since these values are coming from contracts that had to make it through the VM, their values are validated to a certain extent. However, I think it could still be possible for one or more of these values to be falsy but not undefined; for example, what if `amount` was `null`.

TLDR; I think we should check for falsyness instead. Perhaps there's an exception for `""`, in any case where that's a valid value. Perhaps we can also combine this condition with the one that follows.

---

P.S.
If we really needed this strict of a check I we could've factored this out to a helper, for example:
```typescript
function isAnyUndefined(...args: any[]): boolean {
  return args.any(e => typeof(e) === "undefined");
}
```

Then usage would look like:
```typescript
if (isAnyUndefined(amount, fromAddress, contract) {
  ...
}
```

Reply:
I realised that your suggestion to check for falsyness was carried out a few lines beneath - so I removed this unnecessary check